### PR TITLE
Refactor & fix FancyButton

### LINKED ISSUE #40: Component: Text Link
This is essentially just a button without bg. Add example to the storybook

### LINKED ISSUE #61: Fix texture offset when using Graphics
There is an offset when using Graphics to the left and top, that is 50% of the texture

### LINKED ISSUE #62: FancyButton when set `views` shows old textures
For some time it's shown, goese away after rehover

### PR DESCRIPTION
**Pull request by CyberDex**
FancyButton refactored and fixed.

Now it extends `ButtonContainer` to avoid button events reimplementation.
`views` setter removed and dedicated setters are implemented for every view instead.

Fixed some issues related to delayed views update/resize/reposition when views are updated after instance creation.

fixes #62
fixes #61

Implemented example of text link (text button)
closes #40

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -95,7 +95,7 @@
         "jest": "^26.6.3",
         "jest-raw-loader": "^1.0.1",
         "lint-staged": "^13.1.0",
-        "storybook": "7.0.22",
```
Comment (CyberDex) [inline package.json:98]: Storybook update

File: src/Button.ts [hunk 2]
```diff
@@ -32,17 +32,11 @@ export class Button extends ButtonEvents
...
-    }
-
-    /**
-     * Creates and connect interaction events.
-     * @param {Container} view - instance of container, to be turned into button
-     */
-    init(view: Container)
-    {
-        this.view = view;
-        this.enabled = true;
```
Comment (CyberDex) [inline src/Button.ts:45]: Init function was excess here

File: src/Button.ts [hunk 3]
```diff
@@ -53,9 +47,7 @@ export class Button extends ButtonEvents
         if (wasItInitiated) this.disconnectEvents(this._view);
 
         this._view = view;
-        this.connectEvents(view);
-
-        if (!wasItInitiated) this.enabled = true;
```
Comment (CyberDex) [inline src/Button.ts:58]: Enabled setter now is in constructor

File: src/Button.ts [hunk 4]
```diff
@@ -109,33 +101,16 @@ export class Button extends ButtonEvents
...
-    /** Set button view, that all the interaction events are applied to. */
-    // eslint-disable-next-line accessor-pairs
-    override set view(_view: Container)
-    {
-        // Do nothing, because we are the view
-    }
-
-    override get view(): Container
-    {
-        return this;
```
Comment (CyberDex) [inline src/Button.ts:139]: Just a refactor + changes related to `Button` class refactor

File: src/stories/button/ButtonContainerSprite.stories.ts [hunk 5]
```diff
@@ -39,6 +30,17 @@ export const ButtonContainerSprite = ({ size, color, disabled, radius, action }:
...
+    const text = new Text('🤙', { fontSize: 70 });
+
+    text.anchor.set(0.5);
+    text.x = buttonView.width / 2;
+    text.y = buttonView.height / 2;
+
+    buttonView.addChild(text);
+
+    button.addChild(buttonView);
+
```
Comment (CyberDex) [inline src/stories/button/ButtonContainerSprite.stories.ts:43]: Just a refactor to show more clear that ButtonContainer is like a simple container.

File: src/stories/fancyButton/FancyButtonDynamicUpdate.stories.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
+import { FancyButton } from '../../FancyButton';
```
Comment (CyberDex) [inline src/stories/fancyButton/FancyButtonDynamicUpdate.stories.ts:1]: Storybook to show and test dynamic setting of button views and props applied properly.

File: src/FancyButton.ts [hunk 7]
```diff
@@ -99,27 +105,26 @@ export type ButtonOptions = Views & {
...
-    padding: number;
+    _padding: number;
 
     /** Offset of the button state views. If state views have different sizes, this option can help adjust them. */
-    offset: Offset & Pos;
+    _offset: Offset & Pos;
 
     /** Offset of the text view. Can be set to any state of the button. */
-    textOffset: Offset;
+    _textOffset: Offset;
```
Comment (CyberDex) [inline src/FancyButton.ts:121]: Setters implemented

File: src/FancyButton.ts [hunk 8]
```diff
@@ -207,18 +194,16 @@ export class FancyButton extends Container
...
-            disabledView,
-            text,
-            icon
-        };
+        this.defaultView = defaultView;
+        this.hoverView = hoverView;
+        this.pressedView = pressedView;
+        this.disabledView = disabledView;
+        this.textView = text;
+        this.iconView = icon;
```
Comment (CyberDex) [inline src/FancyButton.ts:202]: Refactored to dedicated setters

File: src/FancyButton.ts [hunk 9]
```diff
@@ -275,21 +259,23 @@ export class FancyButton extends Container
      *
      * Plays animations if they are set.
      * @param {State} newState
+     * @param force
      */
-    setState(newState: State)
+    setState(newState: State, force = false)
```
Comment (CyberDex) [inline src/FancyButton.ts:264]: Used to reinvoke state applying in case of view update

File: src/FancyButton.ts [hunk 10]
```diff
@@ -129,27 +131,8 @@ export class FancyButton extends Container
...
-    onUpOut: Signal<(btn?: this, e?: FederatedPointerEvent) => void>;
-    /** Event that fired when the mouse is out of the view */
-    onOut: Signal<(btn?: this, e?: FederatedPointerEvent) => void>;
-    /** Event that is fired when the button is pressed. */
-    onPress: Signal<(btn?: this, e?: FederatedPointerEvent) => void>;
-    /** Event that is fired when the mouse hovers the button. Fired only if device is not mobile.*/
-    onHover: Signal<(btn?: this, e?: FederatedPointerEvent) => void>;
-
-    /**
-     * Turns a given container-based view into a button by adding all button events.
```
Comment (CyberDex) [inline src/FancyButton.ts:152]: Removes as now it is in `ButtonContainer`

File: src/FancyButton.ts [hunk 11]
```diff
@@ -455,179 +436,181 @@ export class FancyButton extends Container
...
-     * If state view is not set (undefined), it will not be changed,
-     * so if it was set before, it will remain the same.
-     * If state view is set to null, it will be removed from the button.
-     * If state view is set it will be updated or added to a button.
-     * @param {Views} views
+     * Sets the default view of the button.
+     * @param { string | Container } view - string (path to the image) or a Container-based view
      */
-    // eslint-disable-next-line accessor-pairs
-    set views(views: Views)
```
Comment (CyberDex) [inline src/FancyButton.ts:466]: Removed setter

File: src/FancyButton.ts [hunk 12]
```diff
@@ -685,65 +668,85 @@ export class FancyButton extends Container
...
+        this.onHover.connect(() =>
+        {
+            if (!this.isDown)
+            {
+                utils.isMobile.any
+                    ? this.setState('default')
+                    : this.setState('hover');
+            }
+        });
+    }
```
Comment (CyberDex) [inline src/FancyButton.ts:714]: State switch logic

**Review by Zyie — APPROVED** [review]

**Pull request merged**